File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int biggest_size = 0;
        for (int i = 0, j = heights.size() - 1; i < j;) {
            int left = heights[i];
            int right = heights[j];
            int size = min(left, right) * (j - i);
            cout << size << endl;
            if (size > biggest_size) {
                biggest_size = size;
            }
            
            if (left > right) {
                j--;
            }
            else {
                i++;
            }

        }
        return biggest_size;   
    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        if not root:
            return 0
        sol = 0
        dq = deque([(root, float('-inf'))])
        while dq:
            curr, curr_max = dq.popleft()
            if curr.val >= curr_max:
                sol += 1
                curr_max = curr.val
            if curr.left:
                dq.append((curr.left, curr_max))
            if curr.right:
                dq.append((curr.right, curr_max))
        return sol